helix-term/src/commands.rs

```diff
@@ -1678,116 +1693,128 @@ fn global_search(cx: &mut Context) {
 
                 let search_root = std::env::current_dir()
                     .expect("Global search error: Failed to get current dir");
-                WalkBuilder::new(search_root)
-                    .hidden(file_picker_config.hidden)
-                    .parents(file_picker_config.parents)
-                    .ignore(file_picker_config.ignore)
-                    .git_ignore(file_picker_config.git_ignore)
-                    .git_global(file_picker_config.git_global)
-                    .git_exclude(file_picker_config.git_exclude)
-                    .max_depth(file_picker_config.max_depth)
-                    .build_parallel()
-                    .run(|| {
-                        let mut searcher_cl = searcher.clone();
-                        let matcher_cl = matcher.clone();
-                        let all_matches_sx_cl = all_matches_sx.clone();
-                        Box::new(move |dent: Result<DirEntry, ignore::Error>| -> WalkState {
-                            let dent = match dent {
-                                Ok(dent) => dent,
-                                Err(_) => return WalkState::Continue,
-                            };
-
-                            match dent.file_type() {
-                                Some(fi) => {
-                                    if !fi.is_file() {
-                                        return WalkState::Continue;
+
+                let search_task = tokio::task::spawn_blocking(move || -> Vec<(usize, PathBuf)> {
+                    // We're in the background thread now so std::sync::mpsc::channel should suffice
+                    let (all_matches_sx, all_matches_rx) = std::sync::mpsc::channel();
+                    WalkBuilder::new(search_root)
```

Comment:
I think we need to move this into another function since there are too many nesting here.